pkg/core/data/ipc/transactions/provider.go

```diff
@@ -190,7 +190,17 @@ func (e *executor) Finalize(ctx context.Context, calls []ContractCall, stateRoot
 		return nil, user.Provisioners{}, nil, err
 	}
 
+	resCalls, err := e.convertToContractCall(res.Txs)
+	if err != nil {
+		return nil, user.Provisioners{}, nil, err
```

Comment:
If we fail to convert txs we should get a fresh provisioners list instead of propagating errors.

This will never happen with the current implementation, but seems safer to me

Reply:
Maybe let's handle this with another issue/PR.  In this PR, I'm just moving the convertToContractCall call a few lines above - before GetProvisioners.